File: zhanglianzhuang/day4/demo04/src/main/java/com/example/demo/code/Command/HttpContextUtil.java
package com.example.demo.code.Command;



import org.springframework.web.context.request.RequestContextHolder;
import org.springframework.web.context.request.ServletRequestAttributes;

import javax.servlet.http.HttpServletRequest;

public class HttpContextUtil {

	//获取请求的对象
	public static HttpServletRequest getRequest() {
		return ((ServletRequestAttributes) RequestContextHolder.getRequestAttributes()).getRequest();
	}

	//获取ip地址
	public static String getIpAddress() {
		String ip = "127.0.0.1";
		HttpServletRequest request = getRequest();
		ip = request.getHeader("x-forwarder-for");
		if (ip == null || ip.length() == 0 || "unknown".equalsIgnoreCase(ip)) {
			ip = request.getHeader("Proxy-Client-IP");
		}
		if (ip == null || ip.length() == 0 || "unknown".equalsIgnoreCase(ip)) {
			ip = request.getHeader("Proxy-Client-IP");
		}
		if (ip == null || ip.length() == 0 || "unknown".equalsIgnoreCase(ip)) {
			ip = request.getHeader("HTTP-CLIENT-IP");
		}
		if (ip == null || ip.length() == 0 || "unknown".equalsIgnoreCase(ip)) {
			ip = request.getHeader("HTTP-X-FORWARDED-FOR");
		}
		if (ip == null || ip.length() == 0 || "unknown".equalsIgnoreCase(ip)) {
			ip = request.getRemoteAddr();
		}

		System.out.println(ip);
		return "0:0:0:0:0:0:0:1".equals(ip) ? "127.0.0.1" : ip;

	}
}

File: caiguicheng/day04/day03作业修改版/demo03work/code/command/HttpContextUtil.kt
package com.example.demo03work.code.command

import org.springframework.web.context.request.RequestContextHolder
import org.springframework.web.context.request.ServletRequestAttributes
import javax.servlet.http.HttpServletRequest


/**
 * Http上下文实用程序类
 * @author 蔡
 * @since 2019-12-26
 * 修改时间:
 */
object HttpContextUtil {

    ///获取请求的对象
    @JvmStatic
    fun getRequest(): HttpServletRequest {
        return (RequestContextHolder.getRequestAttributes() as ServletRequestAttributes).request
    }

    //获取IP地址
    @JvmStatic
    fun getIpAddress(): String {
        var ip :String? = "127.0.0.1"
        val request = getRequest()
        ip = request.getHeader("x-forwarder-for")

        if (ip == null || ip.isEmpty() || "unknown".equals(ip,true)) {
            ip = request.getHeader("Proxy-Client-IP")
        }
        if (ip == null || ip.isEmpty() || "unknown".equals(ip,true)) {
            ip = request.getHeader("WL-Proxy-Client-IP")
        }
        if (ip == null || ip.isEmpty() || "unknown".equals(ip,true)) {
            ip = request.getHeader("HTTP_Client_IP")
        }
        if (ip == null || ip.isEmpty() || "unknown".equals(ip,true)) {
            ip = request.getHeader("HTTP_X_FORWARDED_FOR")
        }
        if (ip == null || ip.isEmpty() || "unknown".equals(ip,true)) {
            ip = request.remoteAddr
        }



        println("ip:$ip")
        return if ("0:0:0:0:0:0:0:1" === ip) "127.0.0.1" else ip
    }
}